catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/model/JsonModelBuilderTest.java

```diff
@@ -0,0 +1,237 @@
+              .setValue("value")
+              .endBinaryComparisonType()
+              .getResult();
+
+      assertThat(node.getType(), is("="));
+      assertThat(node.getNodes(), is(nullValue()));
+      assertThat(node.isLeaf(), is(true));
+      assertThat(FilterLeafNode.class.isInstance(node), is(true));
+
+      FilterLeafNode leaf = (FilterLeafNode) node;
+      assertThat(leaf.getProperty(), is("name"));
+      assertThat(leaf.getValue(), is("value"));
+      assertThat(leaf.isTemplated(), is(false));
+    }
+
+    @Test
+    public void testBinaryComparisonTypeTemplated() {
+      FilterNode node =
+          builder
+              .beginBinaryComparisonType("=")
+              .setProperty("name")
+              .setTemplatedValues("5", "id", true, false)
+              .endBinaryComparisonType()
+              .getResult();
+
+      assertThat(node.getType(), is("="));
+      assertThat(node.getNodes(), is(nullValue()));
+      assertThat(node.isLeaf(), is(true));
+      assertThat(FilterLeafNode.class.isInstance(node), is(true));
+
+      FilterLeafNode leaf = (FilterLeafNode) node;
+      assertThat(leaf.getProperty(), is("name"));
+      assertThat(leaf.getValue(), is(nullValue()));
+      assertThat(leaf.isTemplated(), is(true));
+      assertThat(FilterTemplatedLeafNode.class.isInstance(leaf), is(true));
+
+      FilterTemplatedLeafNode template = (FilterTemplatedLeafNode) leaf;
+      assertThat(template.getDefaultValue(), is("5"));
+      assertThat(template.getNodeId(), is("id"));
+      assertThat(template.isVisible(), is(true));
+      assertThat(template.isReadOnly(), is(false));
+    }
+
+    @Test
+    public void testBinaryComparisonTypeAllOperators() {
+      new JsonModelBuilder().beginBinaryComparisonType("=");
+      new JsonModelBuilder().beginBinaryComparisonType(">");
+      new JsonModelBuilder().beginBinaryComparisonType(">=");
+      new JsonModelBuilder().beginBinaryComparisonType("<");
+      new JsonModelBuilder().beginBinaryComparisonType("<=");
+      new JsonModelBuilder().beginBinaryComparisonType("!=");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testBinaryComparisonTypeBadOperator() {
+      builder.beginBinaryComparisonType("bad");
+    }
+
+    @Test
+    public void testBinaryLogicType() {
+      FilterNode node =
+          builder
+              .beginBinaryLogicType("AND")
+              .beginBinaryComparisonType("=")
+              .setProperty("name")
+              .setValue("value")
+              .endBinaryComparisonType()
+              .endBinaryLogicType()
+              .getResult();
+
+      assertThat("", node.getType(), is("AND"));
+      assertThat("", node.getNodes(), is(not(nullValue())));
+      assertThat("", node.getNodes().size(), is(1));
+    }
+
+    @Test
+    public void testBinaryLogicTypeAllOperators() {
+      new JsonModelBuilder().beginBinaryLogicType("AND");
+      new JsonModelBuilder().beginBinaryLogicType("OR");
```

Comment:
This unit test has no assertions

Reply:
Added a comment to clarify this. No `IllegalArgumentException` indicates a passing test.